BatteryStateOfChargeService.cpp

```diff
@@ -0,0 +1,80 @@
+#include "BatteryStateOfChargeService.h"
+#include "BatteryData.h"
+
+namespace
+{
+    const double BATTERY_AMP_HOUR_CAPACITY = 123.0;
+}
+
+BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
+: initialStateOfChargePercent_(initialStateOfChargePercent)
+{
+    ampHourUsed_ = 0.0;

```

Comment:
Use an [initialization list ](https://en.cppreference.com/w/cpp/language/initializer_list)for stuff in the constructor: 
```
BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
: initialStateOfChargePercent_(initialStateOfChargePercent)
, timeTilCharged(QTime (0,0,0,0))
, timeTilDepleted(QTime (0,0,0,0))
, etc
```

Reply:
Sounds good, I'll double check next time. Not sure how that happened!